apps/remix-ide/src/app/compiler/compiler-utils.js

```diff
@@ -25,10 +25,11 @@ export function canUseWorker (selectedVersion) {
   const version = semver.coerce(selectedVersion)
   const isNightly = selectedVersion.includes('nightly')
   return browserSupportWorker() && (
-    // All compiler versions (including nightlies) after 0.6.3 are wasm compiled
-    semver.gt(version, '0.6.3') ||
-    // Only releases are wasm compiled starting with 0.3.6
-    (semver.gte(version, '0.3.6') && !isNightly)
+    // All compiler versions starting with 0.6.2 (including nightlies) are wasm compiled
+    semver.gte(version, '0.6.2') ||
+    // For older versions wasm builds are only available for official releases
+    // (starting with 0.3.6; for 0.5.17 there is no wasm build yet).
```

Comment:
But the problem with 0.5.17 is the same, if you add a wasm build it will start failing the same way as 0.3.6 did?

What blocks us from releasing 0.5.17?

Reply:
> But the problem with 0.5.17 is the same, if you add a wasm build it will start failing the same way as 0.3.6 did?

Yeah, but don't we already have the opposite problem? Doesn't assuming that 0.5.17 is wasm make remix fail?

I added a comment in our issue to update this here when we have the build: https://github.com/ethereum/solidity/issues/10329#issuecomment-824184647.

> What blocks us from releasing 0.5.17?

Nothing really, it just never was a high priority task. The thing is that someone has to build it and I don't know if @ekpyron's scripts even still work without modifications. Pre-0.6.0 versions require some workarounds so it probably will not be just fire and forget. I still have flashbacks from macOS rebuilds :P

Reply:
Ideally Remix would just always go with the worker option (there's no technical reason why the asmjs can't be run in a worker) :)

Reply:
@axic please confirm if I can merge this

Reply:
@cameel knows the best which versions are available as wasm or not, so I trust his changes and this should be merged as-is. We'll need to remove the exception for 0.5.17 once the wasm build of it is available.

Reply:
> Ideally Remix would just always go with the worker option (there's no technical reason why the asmjs can't be run in a worker) :)

Sure. I made the PR under the assumption that wasm = worker, asm.js = no worker. If that's not true and it's ok to use the worker for asm.js 0.5.17 then I can change it.

Otherwise I think this is what we want until 0.5.17 gets rebuilt as wasm.

Reply:
I think the PR is not working with 0.5.17. have you tried it manually? https://circle-production-customer-artifacts.s3.amazonaws.com/picard/5e74f7eac30084377a70609f/6087dd5d7b9dac370e7e6901-7-build/artifacts/reports/screenshots/remixd.test/Remixd_FAILED_Apr-27-2021-095037-GMT%2B0000-%28Coordinated-Universal.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210427T100756Z&X-Amz-SignedHeaders=host&X-Amz-Expires=60&X-Amz-Credential=AKIAJR3Q6CR467H7Z55A%2F20210427%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Signature=1009f50eba02e0f4a7546832807a22146848d8f31bb2f921c7aa03f614d3c4d4

Reply:
I'm having trouble building it locally (`Could not resolve dependency: npm ERR! peer @babel/core@">=7.0.0-beta.50 <7.0.0-rc.0" from @babel/preset-es2015@7.0.0-beta.53`). Does the CI deploy it somewhere I could access it and try or is a local build the only option?

Anyway I removed the 0.5.17 part from the PR. I thought it was actually broken by being handled the same way as was binaries but if it's not then we can keep it the way it is.

I mean, I know which builds are wasm/asm.js and where they are stored but I am not a JS dev so my ability to tell why it does or does not work with Remix is limited unfortunately.

Reply:
unfortunately, the local build is the only option for external for now. (to have the link you need green tests and for that, you need to create a PR on our repo and have write access.)

ok thanks!

Reply:
https://ipfsgw.komputing.org/ipfs/QmUFUr8wFLtaArCRowyeG78WDnZrFBC33MLdBxMyVVXmMU
 here you can test your changes

Reply:
Sorry, for not getting back to you here earlier. I've been meaning to test it thoroughly to make sure it works but I was off recently and then got some other tasks to deal with first. I see that you went ahead and posted it as a separate PR. Thanks!

BTW, in the end I figured out that one of my issues with a local build was that I needed the `--legacy-peer-deps` option to install dependencies with npm. It appears that there are conflicts between `peerDependencies` and npm v7 is no longer ignoring them by default.